pkg/filesystem/local_directory_windows.go

```diff
@@ -0,0 +1,261 @@
+	}
+	return nil
+}
+
+func PortableSyscallUmask(mask int) () {
+	// No implementation on Windows
+}
+
+// NewLocalDirectory creates a directory handle that corresponds to a
+// local path on the system.
+func NewLocalDirectory(path string) (Directory, error) {
+	path = strings.ReplaceAll(path, "/", string(os.PathSeparator))
+	longAbsPath, err := filepath.Abs(path)
+	if err != nil {
+		return nil, err
+	}
+	if !strings.HasPrefix(longAbsPath, uncPrefix) {
+		longAbsPath = uncPrefix + longAbsPath
+	}
+	return newLocalDirectoryLongAbsPath(longAbsPath)
+}
+
+func newLocalDirectoryLongAbsPath(longAbsPath string) (Directory, error) {
+	fileInfo, err := os.Lstat(longAbsPath)
+	if err != nil {
+		return nil, err
+	}
+	if !fileInfo.IsDir() {
+		return nil, syscall.ENOTDIR
+	}
+	d := &localDirectory{
+		longAbsPath: longAbsPath,
+	}
+	return d, nil
+}
+
+func (d *localDirectory) validateFilenameAndJoin(name string) (string, error) {
+	if err := validateFilename(name); err != nil {
+		return "", err
+	}
+	// path.Join uses the wrong path separator, which is critical for UNC paths. Join manually.
+	return d.longAbsPath + string(os.PathSeparator) + name, nil
+}
+
+func (d *localDirectory) Enter(name string) (Directory, error) {
+	fullPath, err := d.validateFilenameAndJoin(name)
+	if err != nil {
+		return nil, err
+	}
+	return newLocalDirectoryLongAbsPath(fullPath)
+}
+
+func (d *localDirectory) Close() error {
+	return nil
+}
+
+func (d *localDirectory) Link(oldName string, newDirectory Directory, newName string) error {
+	fullOldPath, err := d.validateFilenameAndJoin(oldName)
+	if err != nil {
+		return err
+	}
+
+	d2, ok := newDirectory.(*localDirectory)
+	if !ok {
+		return errors.New("Source and target directory have different types")
+	}
+	fullNewPath, err := d2.validateFilenameAndJoin(newName)
+	if err != nil {
+		return err
+	}
+
+	return os.Link(fullOldPath, fullNewPath)
+}
+
+func (d *localDirectory) Lstat(name string) (FileInfo, error) {
+	fullPath, err := d.validateFilenameAndJoin(name)
+	if err != nil {
+		return nil, err
+	}
+
+	fileInfo, err := os.Lstat(fullPath)
+	if err != nil {
+		return nil, err
+	}
+	mode := fileInfo.Mode()
+	return NewSimpleFileInfo(name, mode), nil
+}
+
+func (d *localDirectory) Mkdir(name string, perm os.FileMode) error {
+	fullPath, err := d.validateFilenameAndJoin(name)
+	if err != nil {
+		return err
+	}
+
+	err = os.Mkdir(fullPath, perm)
+	if err != nil {
+		if fileInfo, err2 := os.Stat(fullPath); err2 == nil && fileInfo.IsDir() {
+			// Already exists, possibly with other casing.
+			return nil
+		}
+	}
+	return err
+}
+
+func (d *localDirectory) OpenFile(name string, flag int, perm os.FileMode) (File, error) {
+	fullPath, err := d.validateFilenameAndJoin(name)
+	if err != nil {
+		return nil, err
+	}
+
+	f, err := os.OpenFile(fullPath, flag, perm)
+	if err != nil {
+		return nil, err
+	}
+	return f, nil
+}
+
+func (d *localDirectory) ReadDir() ([]FileInfo, error) {
+	// Obtain filenames in current directory.
```

Comment:
On UNIX, I had to hand-roll this function, because I needed `*at()` calls. Because you're dealing with pathname strings, I suspect you can just call into https://golang.org/pkg/io/ioutil/#ReadDir ?

Reply:
Though I suspect that this no longer applies in case we alter `FileInfo` to diverge from `os.FileInfo`.